code/utils.py

```diff
@@ -20,12 +25,77 @@ def get_userid(u):
         return None
 
 
+def get_user_emails():
+    return user_id_lookup.keys()
+
+
 def populate_userid_mapping():
     """
-
+    populates the user_id_lookup table based on file employee_id_mapping.csv extracted from the db
     """
     mapping = np.loadtxt('../resources/employee_id_mapping.csv', dtype='str', delimiter=',', skiprows=1)
     for m in mapping:
         user_id_lookup[m[0]] = int(m[1])
 
 
+def plot_with_tsne(email_ids, embeddings, display_plot):
```

Comment:
Do we need a display_plot flag here?
We could set it to have a default True value

Reply:
we need it because mpld3 cannot display multiple plots, but someone may want to run just tSNE after every epoch. Yes, I will put the default value as true